ds3_java_cli/src/main/java/com/spectralogic/ds3cli/ArgumentFactory.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * *****************************************************************************
+ *   Copyright 2014-2016 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ***************************************************************************
+ */
+
+package com.spectralogic.ds3cli;
+
+import ch.qos.logback.classic.Level;
+import com.google.common.collect.ImmutableMap;
+import com.spectralogic.ds3cli.exceptions.BadArgumentException;
+import com.spectralogic.ds3cli.util.Metadata;
+import com.spectralogic.ds3cli.util.Utils;
+import com.spectralogic.ds3client.Ds3Client;
+import com.spectralogic.ds3client.Ds3ClientBuilder;
+import com.spectralogic.ds3client.models.Priority;
+import com.spectralogic.ds3client.models.WriteOptimization;
+import com.spectralogic.ds3client.models.common.Credentials;
+import com.spectralogic.ds3client.utils.Guard;
+import org.apache.commons.cli.*;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+public class ArgumentFactory {
+
+    // always added and read in first parsing
+    public final static Option endpoint = new Option("e", true, "The ds3 endpoint to connect to or have \"DS3_ENDPOINT\" set as an environment variable.");
```

Comment:
All `final static` variable should have the variable name in all caps and separated with `_`